### DISCUSSION TITLE
[V5] Listen to non-numeric port

### DISCUSSION THREAD

**Discussion by thespino** [adonisjs/core#2582] [Ideas]
Hi, i'm hosting an Adonis api server on IIS with iisnode.
Iisnode uses pipe ports, which are strings, to proxy connections. In order for this to work, I always have to edit this line: https://github.com/adonisjs/core/blob/f8ce27f8925f8b5f1ede63f2878b79e6b0c86f8a/src/Ignitor/HttpServer/index.ts#L161 in node_modules removing the cast to number.

Of course it's really bad approach, but I wonder if the cast is really needed. Can it be possible that this will be removed in future releases?

Thanks.

**Comment by thetutlage** [comment]
You don't have to if you validate the environment variable as a number, then AdonisJS will convert it to a number from a string representation of a number. https://docs.adonisjs.com/guides/environment-variables#envschemanumber

**Reply by thespino** [reply]
True, but what I mean is:
On windows there are *named pipe ports* (https://en.wikipedia.org/wiki/Named_pipe#In_Windows) which are special ports used by some programs to communicate between each other, like iisnode does. Those ports aren't numeric, they're strings like "\\.\pipe\foo".

So, the Number cast in `core/src/Ignitor/HttpServer/index.ts` is not valid (returns NaN). That's why I have to edit node_modules each time.

I know it's strange and very specific, that's why I put this in ideas category. In normal situations the Number cast is the expected behaviour.

Thanks a lot, Have a nice day!
